src/client/common/application/customEditorService.ts

```diff
@@ -48,35 +49,54 @@ export class CustomEditorService implements ICustomEditorService, IExtensionSing
         let updateType: 'added' | 'removed' | undefined;
         try {
             const settings = this.workspace.getConfiguration('workbench', undefined);
-            const editorAssociations = settings.get('editorAssociations') as {
-                viewType: string;
-                filenamePattern: string;
-            }[];
+            const editorAssociations = settings.get('editorAssociations') ?? {}; // At this point we don't know if this is the old or new format
 
-            // Update the settings.
-            if (
-                this.useCustomEditorApi &&
-                (editorAssociations.length === 0 || !editorAssociations.find((item) => item.viewType === ViewType))
-            ) {
-                editorAssociations.push({
-                    viewType: ViewType,
-                    filenamePattern: '*.ipynb'
-                });
-                updateType = 'added';
-                await settings.update('editorAssociations', editorAssociations, vscode.ConfigurationTarget.Global);
-                sendTelemetryEvent(Telemetry.UpdateCustomEditorAssociation, undefined, { type: 'added' });
-            }
+            if (Array.isArray(editorAssociations)) {
+                // Old format for editorAssociations setting
+                const hasCustomEditorAssociation =
+                    editorAssociations.find((item) => item.viewType === ViewType) !== undefined;
+
+                // Update the settings.
+                if (this.useCustomEditorApi && (editorAssociations.length === 0 || !hasCustomEditorAssociation)) {
+                    const updatedSettings = [
+                        ...editorAssociations,
+                        {
+                            viewType: ViewType,
+                            filenamePattern: '*.ipynb'
+                        }
+                    ];
+                    updateType = 'added';
+                    await settings.update('editorAssociations', updatedSettings, vscode.ConfigurationTarget.Global);
+                    sendTelemetryEvent(Telemetry.UpdateCustomEditorAssociation, undefined, { type: 'added' });
```

Comment:
Why not just use updateType here?

Reply:
No reason other than this is how the code was originally written 😁

Reply:
So the only reason this variable exists is to have something to pass into the catch block? Seems like a waste of a good variable. :D

Reply:
I believe this is resolved by 076bdb7b6ad676a42629953d4231c9217dd73159.